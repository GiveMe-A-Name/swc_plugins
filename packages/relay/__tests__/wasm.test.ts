import { expect, test } from "vitest";
import { transform } from "@swc/core";
import path from "node:path";
import url from "node:url";

const pluginName = "swc_plugin_relay.wasm";

const transformCode = async (
  code: string,
  options = {
    rootDir: "",
  },
) => {
  return transform(code, {
    jsc: {
      parser: {
        syntax: "ecmascript",
      },
      target: "es2018",
      experimental: {
        plugins: [
          [
            path.join(
              path.dirname(url.fileURLToPath(import.meta.url)),
              "..",
              pluginName,
            ),
            options,
          ],
        ],
      },
    },
    filename: "test.js",
  });
};

test("Should load relay wasm plugin correctly", async () => {
  const input = `const myFragment = graphql\`
  fragment FooFragment on Bar {
    id
  }
\`;
useQuery(graphql\`
  query FooQuery {
    id
  }
\`);
`;
  const { code } = await transformCode(input);
  expect(code).toMatchSnapshot();
});
