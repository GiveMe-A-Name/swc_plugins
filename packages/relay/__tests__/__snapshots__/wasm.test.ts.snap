// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should load relay wasm plugin correctly 1`] = `
"const myFragment = require("./__generated__/FooFragment.graphql.js");
useQuery(require("./__generated__/FooQuery.graphql.js"));
"
`;
